runtime/stdlib/account.go

```diff
@@ -1593,16 +1594,37 @@ func changeAccountContracts(
 			},
 		)
 
+		var legacyContractUpgrade bool
+		// if we are allowing legacy contract upgrades, fall back to the old parser when the new one fails
+		if !ignoreUpdatedProgramParserError(err) && invocation.Interpreter.SharedState.Config.LegacyContractUpgradeEnabled {
+			legacyContractUpgrade = true
+			oldProgram, err = old_parser.ParseProgram(
+				invocation.Interpreter.SharedState.Config.MemoryGauge,
```

Comment:
Maybe refactor `invocation.Interpreter.SharedState.Config` and `invocation.Interpreter.SharedState.Config.MemoryGauge` into variables, and use them in the new code, and in the new parser call above.